pkg/querier/tripperware/queryrange/query_range.go

```diff
@@ -249,8 +254,10 @@ func (prometheusCodec) EncodeRequest(ctx context.Context, r tripperware.Request)
 		}
 	}
 
-	// Always ask gzip to the querier
-	h.Set("Accept-Encoding", "gzip")
+	if c.compression == "snappy" || c.compression == "gzip" {
```

Comment:
Can you make those compressions constant? You can initialize a new type for the codec compression